src/basic/fileio.c

```diff
@@ -691,7 +691,7 @@ static int load_env_file_push(
 }
 
 int load_env_file(FILE *f, const char *fname, const char *newline, char ***rl) {
-        char **m = NULL;
+        char **m = *rl;
```

Comment:
In general we like functions to either succeed or fail atomically. So it'd be better to load the new file, and then `strv_extend_strv` the old one. It's a bit less efficient, but seems cleaner.